Stage

MoonSold opened the pull request:
Валидация внутри сервера есть, что делать дальше не понятно

Commit:
start

Commit:
Добавил валидацию, но вряд ли мне это поможет

Commit:
Скажите, что я делаю не так

anna-plsn commented:
Никогда не комить файлы:
1. С точкой (Например, .idea)
2. Библиотеки (vender, extjs)

Чтобы они никогда сами не комитились, узнай, как записать их в .gitignore
Когда так сделаешь, пропадет конфликт слияния веток

anna-plsn commented on api/Entity/AdminEntity.php:
```diff
@@ -0,0 +1,40 @@
+     * @ORM\Column(type="string",nullable = true,name="adminname",unique=true)
+     * @var string
+     */
+    private $adminname;
```
NIT: Если таблица называется admin, то называть поле adminname нет смысла. Лучше просто name

anna-plsn commented on api/Entity/QuestionGroupEntity.php:
```diff
@@ -0,0 +1,41 @@
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="questionsgroup")
```
NIT: questionsgroup. Слова лучше визуально разделять

anna-plsn commented on api/service/AdminUserClass.php:
```diff
@@ -0,0 +1,8 @@
+
+namespace service;
+
+class AdminUserClass
```
AdminUserController

MoonSold replied:
Это не контроллер, это сервис. Будет соответствующая функция в котроллере работающая с этим классом

anna-plsn commented on api/service/AuthUserClass.php:
```diff
@@ -0,0 +1,34 @@
+
+class AuthUserClass
+{
+    public static function authUser($login,$password):array
```
psr

anna-plsn commented on api/service/AuthUserClass.php:
```diff
@@ -0,0 +1,34 @@
+        try {
+            global $entityManager;
+            $user = $entityManager->getRepository(Entity\UsersEntity::class)->findOneBy(array('login' => $login));
+            if ($user->getPasswordHash() == md5($password)) {
```
Через строгое сравнение

anna-plsn commented on api/service/AuthUserClass.php:
```diff
@@ -0,0 +1,34 @@
+{
+    public static function authUser($login,$password):array
+    {
+        try {
```
try catch не нужен, если это select
 Просто проверку результата БД на null и тогда return ["auth"=> false,"token" => "not auth", "username" => "not auth"];

anna-plsn commented on api/service/RegisterUserClass.php:
```diff
@@ -0,0 +1,43 @@
+    {
+        try {
+            $validation = ValidatorClass::registerValidator($login, $password, $username);
+            if ($validation === true) {
```
if ($validation) {

anna-plsn commented on api/service/ValidatorClass.php:
```diff
@@ -0,0 +1,18 @@
+        if (preg_match("/^[a-zA-Z0-9-_]{5,20}$/i",$login)
+                && preg_match("/^[a-zA-Z0-9-_#$^&*@!?]{5,20}$/i",$password)
+                    && preg_match("/^[а-яёА-ЯЁ]+$/u",$username)){
+                return true;
```
```
return preg_match("/^[a-zA-Z0-9-_]{5,20}$/i",$login)
                && preg_match("/^[a-zA-Z0-9-_#$^&*@!?]{5,20}$/i",$password)
                    && preg_match("/^[а-яёА-ЯЁ]+$/u",$username);
```

MoonSold replied:
По факту

anna-plsn commented on index.php:
```diff
@@ -0,0 +1,28 @@
+require_once 'api/vendor/autoload.php';
+require_once 'api/controller/controller.php';
+
+register_shutdown_function(function () {
```
Это не комитят, это только для отладки

MoonSold replied:
я понял, забыл убрать

Commit:
ok

Commit:
type: fix
subject: Изменен api
body: Изменена структура api, добавлен файл api.php, обрабабтывающий запросы от клиента и вызывающий соответствующий метод

MoonSold commented:
Ext js странный

Commit:
type:fix
scope: front
Изменён инструмент для написания фронта
Фронт теперь пишится на EXT JS, запросы от сервера доходят и могут быть обработаны, сервер выдаёт нужные данные из БД. Не работает подключение контроллеров и элементов к app.js. Нужна доработка структуры фронта. Требуется помощь с пониманием EXT.js

Commit:
type: feat
Добавлены новые функции
Реализованы новые функции регистрации и авторизации пользователей, но куки обновляются на фроне, что напрягает. Непонятно как работать с хранилищем и рендерингом аккордеона, а также удаление и создание кнопок на тулбаре при регистрации и авторизации. Требуется помощь.

Commit:
type:fix Добавление админки и почти доделование юзерской страницы
Теперь пользователь может голосовать, в данной ветке осталось только добавить функцию запомни меня и голосования только один раз, начало разработки админки.

anna-plsn commented on api/api.php:
```diff
@@ -0,0 +1,32 @@
+        echo json_encode($response);
+        }
+}
+else{
```
elseif ($_REQUEST["actor"] === "admin")

anna-plsn commented on api/controller/AdminController.php:
```diff
@@ -0,0 +1,62 @@
+        $this->adminService = $adminService;
+    }
+
+    public function controllerAdminAuthFunction(array $request): array
```
adminAuth
И так видно, что это функция в контроллере

anna-plsn commented on api/service/AdminService.php:
```diff
@@ -0,0 +1,33 @@
+
+    public function authAdmin(string $login, string $password):array
+    {
+        register_shutdown_function(function () {
```
Средства для дебага не коммитят

MoonSold committed:
Delete .idea directory
